test/parallel/test-worker-prof.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const fs = require('fs');
+const assert = require('assert');
+const { spawnSync } = require('child_process');
+const { Worker } = require('worker_threads');
+
+if (!common.isMainThread)
+  common.skip('process.chdir is not available in Workers');
+
+// Test that --prof also tracks Worker threads.
+// Refs: https://github.com/nodejs/node/issues/24016
+
+if (process.argv[2] === 'child') {
+  const spin = `
+  const start = Date.now();
+  while (Date.now() - start < 200);
```

Comment:
IIUC the profiler breaks on AIX. Moreover, in systems with varying load situations, can this calculation prove flaky? for example, if the time required for these computations and the OS scheduler's turnaround time for a thread to come back are in the order of ~100 millis, and / or if the thread gets de-scheduled between the two lines, we end up not having enough ticks?

So: either increase the spin time to nullify the environmental factors, or bring-in the overall runtime into the picture: say, r/t = K where r is the overall runtime, t is the number of ticks and K is an emperical constant? (and skip for AIX)

Reply:
> IIUC the profiler breaks on AIX.

This might be a dumb question, but: Why did the test pass CI on AIX then?

> Moreover, in systems with varying load situations, can this calculation prove flaky?

I’ve tried to leave some space in the calculation for this to work out okay (one full order of magnitude for my machine). If we do change something, I’d prefer to run a stress test first to check *whether* this is flaky in the real world?

Reply:
I ran this patch and saw it breaks in AIX with `OOM`. In addition, I saw this in some other test:
https://github.com/nodejs/node/blob/ba4df925eb7143606d5a57f49e4ecb179dd7743b/test/parallel/test-tick-processor-arguments.js#L12-L13

hence the proposal.But I now see that it actually passed in AIX CI, so pls ignore this.

Reply:
@gireeshpunathil Which patch OOM’ed? This PR? That shouldn’t be happening as this doesn’t continuously allocate data or something to that effect…

Reply:
it was when I ran `node --prof-process <isolate..v8.log>`. Looks like I am unable to reproduce it!

Reply:
@gireeshpunathil Maybe it’s `--prof-process` and not `--prof` that’s broken on AIX?

Reply:
yes, right!